plugins/products.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Store} from 'redux';
+
+import store from 'stores/redux_store';
+
+import PluginRegistry from './registry';
+
+export abstract class ProductPlugin {
+    abstract initialize(registry: PluginRegistry, store: Store): void;
+    abstract uninitialize(): void;
+}
+
+export function initializeProducts() {
+    /* eslint-disable no-console, import/no-unresolved */
+    return async (/*dispatch, getState*/) => {
+        /**
+         * products contains a map of product IDs to a function that will load all of their parts. Calling that
+         * function will return an object where each field is a Promise that will resolve to that module.
+         *
+         * Note that these import paths must be statically defined or else they won't be found at runtime.
+         *
+         * Since this file doesn't use TypeScript yet, the type definition for it is:
+         *
+         * const products: Array<{
+         *     id: string;
+         *     load: () => {
+         *         index: Promise<{default: ProductPlugin}>;
+         *         manifest: Promise<{default: PluginManifest}>;
+         *     };
+         * }>;
+         */
+        const products = [
+            {
+                id: 'focalboard',
+                load: () => ({
+                    index: import('focalboard'),
+                    manifest: import('focalboard/manifest'),
+                }),
+            },
+        ];
+
+        await Promise.all(products.map(async (product) => {
+            if (!REMOTE_MODULES[product.id]) {
+                console.log(`Product ${product.id} not found. Not loading it.`);
+                return;
+            }
+
+            console.log(`Loading product ${product.id}...`);
+
+            try {
+                // Start loading the product
+                const imports = product.load();
```

Comment:
This runs the imports for each module. We could theoretically load these in separate pieces if we wanted to load them at separate times, but we're just loading them all at once for now